### PR TITLE
Remove syntax.ParseJenkinsfileYaml and fix flaky tests

### PR DESCRIPTION
**Pull request by dwnusbaum**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Description

Follow-up to #3017 and #3026. Removes an exported method used only in tests, replaces its use with `LoadProjectConfig` and adjusts test file layout to match what is expected by `LoadProjectConfig`. Also fixes some flaky tests (map iteration order is nondeterministic) and does some minor cleanup to ruse `filepath.Join` instead of with manually appending strings with `/` to hopefully avoid issues running tests on Windows.

CC @abayer @joseblas

### CODE REVIEW

File: pkg/kpipelines/syntax/pipeline_internal_test.go [hunk 1]
```diff
@@ -54,59 +53,3 @@ func TestFindDuplicates(t *testing.T) {
 		})
 	}
 }
-
-func TestFindDuplicatesWithStages(t *testing.T) {
```
Comment (dwnusbaum) [inline pkg/kpipelines/syntax/pipeline_internal_test.go]: This was doing the same thing as TestValidationFailures, so I just moved the files around and merged them into the other test.

File: pkg/kpipelines/syntax/pipeline.go [hunk 2]
```diff
@@ -1013,18 +1008,19 @@ func findDuplicates(names []string) *apis.FieldError {
...
 		if v > 1 {
-			errors = append(errors, "Duplicate stage name '"+k+"'")
+			duplicateNames = append(duplicateNames, "'"+k+"'")
 		}
 	}
 
-	if len(errors) > 0 {
+	if len(duplicateNames) > 0 {
+		// Avoid nondeterminism in error messages
+		sort.Strings(duplicateNames)
```
Comment (dwnusbaum) [inline pkg/kpipelines/syntax/pipeline.go]: First flaky test fix.

File: pkg/kpipelines/syntax/pipeline.go [hunk 3]
```diff
@@ -524,8 +512,15 @@ func scopedEnv(s Stage, parentEnv []corev1.EnvVar) []corev1.EnvVar {
...
 	env := make([]corev1.EnvVar, 0, len(envMap))
 
-	for _, value := range envMap {
-		env = append(env, value)
+	// Avoid nondeterministic results by sorting the keys and appending vars in that order.
+	var envVars []string
+	for k := range envMap {
+		envVars = append(envVars, k)
+	}
+	sort.Strings(envVars)
```
Comment (dwnusbaum) [inline pkg/kpipelines/syntax/pipeline.go]: Second flaky test fix.

**Comment by abayer** [comment]
/lgtm

Thank you so much for noticing the fix for this before anyone else noticed we'd forgot the flakiness. =)

cc @kshultzCB

**Pull request merged**